freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/KerberosMgmt/KerberosMgmtV1Endpoint.java

```diff
@@ -0,0 +1,50 @@
+package com.sequenceiq.freeipa.api.v1.KerberosMgmt;
+
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.doc.KeytabModelNotes;
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.doc.KeytabOperationsDescription;
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.model.ServiceKeytabRequest;
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.model.ServiceKeytabResponse;
+import com.sequenceiq.service.api.doc.ContentType;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+
+@Path("/v1/kerberosmgmt")
+@Consumes(MediaType.APPLICATION_JSON)
+@Api(value = "/v1/kerberosmgmt", protocols = "http,https")
+public interface KerberosMgmtV1Endpoint {
+
+    @POST
+    @Path("createprincipal")
```

Comment:
I think this post API should be `@Path("servicekeytab")` and the POST will create the initial create of the host, create the principal, and generate a keytab (this is not an input parameter but rather something that is randomly generated on the server side). So the POST in this case will also return the keytab which is generated.

I think what I am asking for somewhat deviates from the "normal" web POST in the sense that not all of the parameters will be provided as input to the POST (ie. the keytab is not an input parameter but rather it is generated as a part of the process and then returned as an output).


Side note 1:
The GET `@Path("servicekeytab")`: The GET will retrieve an existing keytab. It has to be preexisting from a prior POST. This is because of constraints from the `ipa-getkeytab` program. The `ipa-getkeytab` program has a `--retrieve` option that can only be used for service principals that have already been created and the service principal already has a keytab. This prevents the keytab from being modified when retrieving it and it creates 2 code paths:
Path 1: Creates a new keytab which invalidates old keytabs
Path 2 ("--retrieve"): The existing keytab is returned without modification. Prior keytabs are not invalidated.

`ipa-getkeytab` man page
```
       WARNING: retrieving the keytab resets the secret for the Kerberos prin‐
       cipal.   This  renders  all  other  keytabs for that principal invalid.
       When multiple hosts or  services  need  to  share  the  same  key  (for
       instance  in  high  availability  or  load  balancing clusters), the -r
       option must be used to retrieve the existing key instead of  generating
       a new one (please refer to the EXAMPLES section).
```

Side note 2:
I think the 1 GET and 2 DELETE API paths are good.

Reply:
This is not what we discussed before. This approach looks new.

Reply:
done.